libarchive/archive_read_support_format_tar.c

```diff
@@ -2546,20 +2559,16 @@ pax_attribute(struct archive_read *a, struct tar *tar, struct archive_entry *ent
 			else if (key_length > 6 && memcmp(key, "xattr.", 6) == 0) {
 				key_length -= 6;
 				key += 6;
-				if (value_length > xattr_limit) {
+				p = __archive_read_ahead(a, value_length, &bytes_read);
```

Comment:
This code was originally trying to protect against a malicious archive that:
* had a `value_length` here that was rediculously large (e.g., 32GiB)
* but the archive was actually truncated well before then.

With the code you have here, libarchive will allocate a buffer to store that data before even trying to read the data into memory, likely crashing the host application in the process.  That's why I really wanted to put some limit here.  I'm not surprised that 16MiB turned out to be too small, but you're effectively raising that limit to 2^64, which seems excessive.